### PR TITLE
Corrected the issues preventing the package from being built on graviton

### PR DESCRIPTION
**Pull request by varlackc**
Made some modifications to the recipe in order to build the package on graviton.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
+# Get an updated config.sub and config.guess
+cp $BUILD_PREFIX/share/gnuconfig/config.* .
 
-export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS}"
+# export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS}"
+# The flag -Wl,--allow-multiple-definition solves issues with libraries needed to build the package
+export LDFLAGS="-L${PREFIX}/lib ${LDFLAGS}-Wl,--allow-multiple-definition"
```
Comment (chenghlee) [inline recipe/build.sh:7]: 1. There needs to be a space between `${LDFLAGS}` and `-Wl,--allow-multiple-definition`.
2. More importantly, using `-Wl,--allow-multiple-definition` seems like a _really_ bad idea; it allows the definition of one symbol (function) to squash another. Is there some reference that says it should be used for this build?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -29,8 +29,11 @@ requirements:
     - automake  # [linux]
     - libtool   # [linux]
     - pkg-config  # [not win]
+    # Used gnucofig and updated config.sub and config.guess files
+    - gnuconfig  # [unix]
     - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
+    # We were getting an error over depending, there was no c++ code
+    # - {{ compiler('cxx') }}
```
Comment (chenghlee) [inline recipe/meta.yaml:36]: The presence of this dependency strongly suggests that someone expects C++ bindings, but the build script somehow did not generate them; stripping this out may potentially change the behavior of downstream packages. I would look into the output of `./configure --help`, the upstream docs, etc. to figure out if C++ bindings are even an option, and if so, why this recipe is not configuring and/or building it.

File: recipe/build.sh (hunk 1 above)
Comment (chenghlee) [inline recipe/build.sh:7]: There needs to be a space between ${LDFLAGS} and -Wl,--allow-multiple-definition.
More importantly, using -Wl,--allow-multiple-definition seems like a really bad idea; it allows the definition of one symbol (function) to squash another. Is there some reference that says it should be used for this build?

File: recipe/meta.yaml (hunk 2 above)
Comment (chenghlee) [inline recipe/meta.yaml:36]: The presence of this dependency strongly suggests that someone expects C++ bindings, but the build script somehow did not generate them; stripping this out may potentially change the behavior of downstream packages. I would look into the output of `./configure --help`, the upstream docs, etc. to figure out if C++ bindings are even an option, and if so, why this recipe is not configuring and/or building it.

**Review by chenghlee — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (chenghlee) [reply recipe/meta.yaml:36]: Also: if you're going to remove a dependency from a recipe, you need to re-build & test it on the other supported platforms to make sure you haven't unexpectedly broken something.

File: recipe/build.sh (hunk 1 above)
Comment (katietz) [reply recipe/build.sh:7]: well, it depends if that option is good, or not.  In general I agree, but there are projects which are redefining symbols in various object files and assume that they get a weak linkage ... in such cases this option can be pretty useful

Comment (chenghlee) [reply recipe/build.sh:7]: I'm unable replicate the multiple definition error when building in a freshly-started and configured container, nor does it seem to be happening on x86_64 with GCC 7.x, 8.x, or 9.x. The only thing I see in the [GCC 10 changelog](https://gcc.gnu.org/gcc-10/changes.html#c) that might trigger this issue is switching default behavior from `-fcommon` to `-fno-common`, but I'm not convinced that that is the root cause of the reported problem.

@varlackc: can you take this LDFLAGS line out of the build.sh, then trying building this package in a freshly-started container?

File: recipe/meta.yaml (hunk 2 above)
Comment (chenghlee) [reply recipe/meta.yaml:36]: The over-depending error is happening on x86_64 as well, so I think removing the C++ compiler is fine.

File: recipe/build.sh (hunk 1 above)
Comment (varlackc) [reply recipe/build.sh:7]: @chenghlee : the recommended changes were made. After testing we see that the package build properly for graviton.

**Comment by varlackc** [comment]
The issue was due to a miss-configuration on the local environment. 
The PR is no longer needed.

**Pull request closed without merging**